types/src/write_set.rs

```diff
@@ -249,6 +246,20 @@ impl TransactionWrite for WriteOp {
             Deletion | DeletionWithMetadata { .. } => (),
         }
     }
+
+    fn as_modification(&self) -> Option<Self> {
+        use WriteOp::*;
+
+        match self {
+            Creation(data) | Modification(data) => Some(Modification(data.clone())),
+            CreationWithMetadata { data, metadata }
+            | ModificationWithMetadata { data, metadata } => Some(ModificationWithMetadata {
+                data: data.clone(),
+                metadata: metadata.clone(),
+            }),
+            _ => None,
```

Comment:
Seems like we can just keep deletion a deletion? Given that we preserve modifications

Reply:
maybe I should rename the method, but this is WriteOp from read that needs to be exchanged - with identifier that we modified inside it

Deletion cannot have exchange.
but even just logically - as_modification should only return modification when possible. create and modifcation can be converted to modification, but deletion cannot - so returning deletion doesn't make sense in that context either

Reply:
Having a bit more descriptive name would be great